src/components/core/GeoChart.tsx

```diff
@@ -65,70 +71,77 @@ export interface IGeoChartInnerOptions {
 export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
+        documentObj: document,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPosition: TOP,
         config: {
             mapboxToken: "",
         },
     };
 
     private geoChartOptions: IGeoChartInnerOptions;
+    private throttledOnWindowResize: any;
 
     public constructor(props: IGeoChartInnerProps) {
         super(props);
         this.state = {
             enabledLegendItems: [],
+            showFluidLegend: this.shouldShowFluid(),
         };
+        this.throttledOnWindowResize = throttle(this.onWindowResize.bind(this), 100);
+        this.validateAndUpdateData();
```

Comment:
Why does we need `validateAndUpdateData` at here ? When component is initialization the `Execution` is `null` and we should not use `setState` in `constructor`

Reply:
fixed. thank you :)